{"matches":[{"vulnerability":{"id":"GHSA-5rjg-fvgr-3xxf","dataSource":"https://github.com/advisories/GHSA-5rjg-fvgr-3xxf","namespace":"github:language:python","severity":"High","urls":[],"description":"setuptools has a path traversal vulnerability in PackageIndex.download that leads to Arbitrary File Write","cvss":[],"epss":[{"cve":"CVE-2025-47273","epss":0.00435,"percentile":0.62062,"date":"2025-06-03"}],"fix":{"versions":["78.1.1"],"state":"fixed"},"advisories":[],"risk":0.32625},"relatedVulnerabilities":[{"id":"CVE-2025-47273","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2025-47273","namespace":"nvd:cpe","severity":"High","urls":["https://github.com/pypa/setuptools/blob/6ead555c5fb29bc57fe6105b1bffc163f56fd558/setuptools/package_index.py#L810C1-L825C88","https://github.com/pypa/setuptools/commit/250a6d17978f9f6ac3ac887091f2d32886fbbb0b","https://github.com/pypa/setuptools/issues/4946","https://github.com/pypa/setuptools/security/advisories/GHSA-5rjg-fvgr-3xxf","https://lists.debian.org/debian-lts-announce/2025/05/msg00035.html","https://github.com/pypa/setuptools/issues/4946"],"description":"setuptools is a package that allows users to download, build, install, upgrade, and uninstall Python packages. A path traversal vulnerability in `PackageIndex` is present in setuptools prior to version 78.1.1. An attacker would be allowed to write files to arbitrary locations on the filesystem with the permissions of the process running the Python code, which could escalate to remote code execution depending on the context. Version 78.1.1 fixes the issue.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:N/AC:L/AT:N/PR:N/UI:N/VC:N/VI:H/VA:N/SC:N/SI:N/SA:N/E:P/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":7.7},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-47273","epss":0.00435,"percentile":0.62062,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"setuptools","version":"75.1.0"}},"found":{"versionConstraint":"<78.1.1 (python)","vulnerabilityID":"GHSA-5rjg-fvgr-3xxf"},"fix":{"suggestedVersion":"78.1.1"}}],"artifact":{"id":"4dab81cde6bb3286","name":"setuptools","version":"75.1.0","type":"python","locations":[{"path":"/lib/python3.10/site-packages/setuptools-75.1.0.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/setuptools-75.1.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/setuptools-75.1.0.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/setuptools-75.1.0.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/setuptools-75.1.0.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/setuptools-75.1.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["MIT"],"cpes":["cpe:2.3:a:python:setuptools:75.1.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/setuptools@75.1.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-53q9-r3pm-6pq6","dataSource":"https://github.com/advisories/GHSA-53q9-r3pm-6pq6","namespace":"github:language:python","severity":"Critical","urls":[],"description":"PyTorch: `torch.load` with `weights_only=True` leads to remote code execution","cvss":[],"epss":[{"cve":"CVE-2025-32434","epss":0.0032,"percentile":0.54534,"date":"2025-06-03"}],"fix":{"versions":["2.6.0"],"state":"fixed"},"advisories":[],"risk":0.28800000000000003},"relatedVulnerabilities":[{"id":"CVE-2025-32434","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2025-32434","namespace":"nvd:cpe","severity":"Critical","urls":["https://github.com/pytorch/pytorch/security/advisories/GHSA-53q9-r3pm-6pq6"],"description":"PyTorch is a Python package that provides tensor computation with strong GPU acceleration and deep neural networks built on a tape-based autograd system. In version 2.5.1 and prior, a Remote Command Execution (RCE) vulnerability exists in PyTorch when loading a model using torch.load with weights_only=True. This issue has been patched in version 2.6.0.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":9.8,"exploitabilityScore":3.9,"impactScore":5.9},"vendorMetadata":{}},{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:N/AC:L/AT:N/PR:N/UI:N/VC:H/VI:H/VA:H/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":9.3},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-32434","epss":0.0032,"percentile":0.54534,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"torch","version":"2.4.1+cpu"}},"found":{"versionConstraint":"<=2.5.1 (python)","vulnerabilityID":"GHSA-53q9-r3pm-6pq6"},"fix":{"suggestedVersion":"2.6.0"}}],"artifact":{"id":"c15427c9f84b45c1","name":"torch","version":"2.4.1+cpu","type":"python","locations":[{"path":"/lib/python3.10/site-packages/torch-2.4.1+cpu.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/torch-2.4.1+cpu.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/torch-2.4.1+cpu.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/torch-2.4.1+cpu.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/torch-2.4.1+cpu.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/torch-2.4.1+cpu.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3"],"cpes":["cpe:2.3:a:pytorch_team_project:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_team_project:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_teamproject:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_teamproject:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packages_project:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packages_project:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packagesproject:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packagesproject:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_team_project:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python-torch:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python-torch:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python_torch:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python_torch:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_team:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_team:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_teamproject:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packages_project:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packages:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packages:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packagesproject:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python-torch:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python_torch:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_team:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:torch:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:torch:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packages:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:torch:torch:2.4.1+cpu:*:*:*:*:*:*:*"],"purl":"pkg:pypi/torch@2.4.1%2Bcpu","upstreams":[]}},{"vulnerability":{"id":"GHSA-44wm-f244-xhp3","dataSource":"https://github.com/advisories/GHSA-44wm-f244-xhp3","namespace":"github:language:python","severity":"High","urls":[],"description":"Pillow buffer overflow vulnerability","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:H/PR:L/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":6.7,"exploitabilityScore":0.8,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-28219","epss":0.00095,"percentile":0.27869,"date":"2025-06-03"}],"fix":{"versions":["10.3.0"],"state":"fixed"},"advisories":[],"risk":0.06745},"relatedVulnerabilities":[{"id":"CVE-2024-28219","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-28219","namespace":"nvd:cpe","severity":"Medium","urls":["https://lists.debian.org/debian-lts-announce/2024/04/msg00008.html","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/4XLPUT3VK4GQ6EVY525TT2QNUIXNRU5M/","https://pillow.readthedocs.io/en/stable/releasenotes/10.3.0.html#security","https://lists.debian.org/debian-lts-announce/2024/04/msg00008.html","https://lists.fedoraproject.org/archives/list/package-announce%40lists.fedoraproject.org/message/4XLPUT3VK4GQ6EVY525TT2QNUIXNRU5M/","https://pillow.readthedocs.io/en/stable/releasenotes/10.3.0.html#security"],"description":"In _imagingcms.c in Pillow before 10.3.0, a buffer overflow exists because strcpy is used instead of strncpy.","cvss":[{"source":"cve@mitre.org","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:H/PR:L/UI:R/S:U/C:H/I:H/A:H","metrics":{"baseScore":6.7,"exploitabilityScore":0.8,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-28219","epss":0.00095,"percentile":0.27869,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"pillow","version":"10.2.0"}},"found":{"versionConstraint":"<10.3.0 (python)","vulnerabilityID":"GHSA-44wm-f244-xhp3"},"fix":{"suggestedVersion":"10.3.0"}}],"artifact":{"id":"b2689639dd7bbdb7","name":"pillow","version":"10.2.0","type":"python","locations":[{"path":"/lib/python3.10/site-packages/pillow-10.2.0.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/pillow-10.2.0.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/pillow-10.2.0.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/pillow-10.2.0.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/pillow-10.2.0.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/pillow-10.2.0.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["HPND"],"cpes":["cpe:2.3:a:\"jeffrey_a__clark_(alex)\"_<aclark_project:python-pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:\"jeffrey_a__clark_(alex)\"_<aclark_project:python_pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:\"jeffrey_a__clark_(alex)\"_<aclarkproject:python-pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:\"jeffrey_a__clark_(alex)\"_<aclarkproject:python_pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:\"jeffrey_a__clark_(alex)\"_<aclark_project:pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:\"jeffrey-a--clark-(alex)\"-<aclark:python-pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:\"jeffrey-a--clark-(alex)\"-<aclark:python_pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:\"jeffrey_a__clark_(alex)\"_<aclark:python-pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:\"jeffrey_a__clark_(alex)\"_<aclark:python_pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:\"jeffrey_a__clark_(alex)\"_<aclarkproject:pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:\"jeffrey-a--clark-(alex)\"-<aclark:pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:\"jeffrey_a__clark_(alex)\"_<aclark:pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python-pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:python_pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python-pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:python_pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python-pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:python_pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python-pillow:pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python-pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python:python_pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python_pillow:pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:pillow:pillow:10.2.0:*:*:*:*:*:*:*","cpe:2.3:a:python:pillow:10.2.0:*:*:*:*:*:*:*"],"purl":"pkg:pypi/pillow@10.2.0","upstreams":[]}},{"vulnerability":{"id":"GHSA-q2x7-8rv6-6q7h","dataSource":"https://github.com/advisories/GHSA-q2x7-8rv6-6q7h","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Jinja has a sandbox breakout through indirect reference to format method","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":7.8,"exploitabilityScore":1.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-56326","epss":0.00103,"percentile":0.29452,"date":"2025-06-03"}],"fix":{"versions":["3.1.5"],"state":"fixed"},"advisories":[],"risk":0.06592},"relatedVulnerabilities":[{"id":"CVE-2024-56326","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-56326","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pallets/jinja/commit/48b0687e05a5466a91cd5812d604fa37ad0943b4","https://github.com/pallets/jinja/releases/tag/3.1.5","https://github.com/pallets/jinja/security/advisories/GHSA-q2x7-8rv6-6q7h"],"description":"Jinja is an extensible templating engine. Prior to 3.1.5, An oversight in how the Jinja sandboxed environment detects calls to str.format allows an attacker that controls the content of a template to execute arbitrary Python code. To exploit the vulnerability, an attacker needs to control the content of a template. Whether that is the case depends on the type of application using Jinja. This vulnerability impacts users of applications which execute untrusted templates. Jinja's sandbox does catch calls to str.format and ensures they don't escape the sandbox. However, it's possible to store a reference to a malicious string's format method, then pass that to a filter that calls it. No such filters are built-in to Jinja, but could be present through custom filters in an application. After the fix, such indirect calls are also handled by the sandbox. This vulnerability is fixed in 3.1.5.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:L/AC:L/AT:P/PR:L/UI:P/VC:H/VI:H/VA:H/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":5.4},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":7.8,"exploitabilityScore":1.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-56326","epss":0.00103,"percentile":0.29452,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"jinja2","version":"3.1.3"}},"found":{"versionConstraint":"<=3.1.4 (python)","vulnerabilityID":"GHSA-q2x7-8rv6-6q7h"},"fix":{"suggestedVersion":"3.1.5"}}],"artifact":{"id":"8ef17ffb7cb9e30e","name":"jinja2","version":"3.1.3","type":"python","locations":[{"path":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:python-jinja2:python-jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python_jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python-jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python_jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python-jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python_jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python-jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python_jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python:jinja2:3.1.3:*:*:*:*:*:*:*"],"purl":"pkg:pypi/jinja2@3.1.3","upstreams":[]}},{"vulnerability":{"id":"GHSA-h75v-3vvj-5mfj","dataSource":"https://github.com/advisories/GHSA-h75v-3vvj-5mfj","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Jinja vulnerable to HTML attribute injection when passing user input as keys to xmlattr filter","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:L/I:L/A:N","metrics":{"baseScore":5.4,"exploitabilityScore":2.9,"impactScore":2.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-34064","epss":0.00085,"percentile":0.25885,"date":"2025-06-03"}],"fix":{"versions":["3.1.4"],"state":"fixed"},"advisories":[],"risk":0.0442},"relatedVulnerabilities":[{"id":"CVE-2024-34064","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-34064","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pallets/jinja/commit/0668239dc6b44ef38e7a6c9f91f312fd4ca581cb","https://github.com/pallets/jinja/security/advisories/GHSA-h75v-3vvj-5mfj","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/567XIGSZMABG6TSMYWD7MIYNJSUQQRUC/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/GCLF44KY43BSVMTE6S53B4V5WP3FRRSE/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/SSCBHIL6BYKR5NRCBXP4XMP2CEEKGFVS/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/ZALNWE3TXPPHVPSI3AZ5CTMSTAVN5UMS/","https://github.com/pallets/jinja/commit/0668239dc6b44ef38e7a6c9f91f312fd4ca581cb","https://github.com/pallets/jinja/security/advisories/GHSA-h75v-3vvj-5mfj","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/567XIGSZMABG6TSMYWD7MIYNJSUQQRUC/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/GCLF44KY43BSVMTE6S53B4V5WP3FRRSE/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/SSCBHIL6BYKR5NRCBXP4XMP2CEEKGFVS/","https://lists.fedoraproject.org/archives/list/package-announce@lists.fedoraproject.org/message/ZALNWE3TXPPHVPSI3AZ5CTMSTAVN5UMS/"],"description":"Jinja is an extensible templating engine. The `xmlattr` filter in affected versions of Jinja accepts keys containing non-attribute characters. XML/HTML attributes cannot contain spaces, `/`, `>`, or `=`, as each would then be interpreted as starting a separate attribute. If an application accepts keys (as opposed to only values) as user input, and renders these in pages that other users see as well, an attacker could use this to inject other attributes and perform XSS. The fix for CVE-2024-22195 only addressed spaces but not other characters. Accepting keys as user input is now explicitly considered an unintended use case of the `xmlattr` filter, and code that does so without otherwise validating the input should be flagged as insecure, regardless of Jinja version. Accepting _values_ as user input continues to be safe. This vulnerability is fixed in 3.1.4.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:L/I:L/A:N","metrics":{"baseScore":5.4,"exploitabilityScore":2.9,"impactScore":2.6},"vendorMetadata":{}},{"source":"134c704f-9b21-4f2e-91b3-4a467353bcc0","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:N/UI:R/S:U/C:L/I:L/A:N","metrics":{"baseScore":5.4,"exploitabilityScore":2.9,"impactScore":2.6},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-34064","epss":0.00085,"percentile":0.25885,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"jinja2","version":"3.1.3"}},"found":{"versionConstraint":"<3.1.4 (python)","vulnerabilityID":"GHSA-h75v-3vvj-5mfj"},"fix":{"suggestedVersion":"3.1.4"}}],"artifact":{"id":"8ef17ffb7cb9e30e","name":"jinja2","version":"3.1.3","type":"python","locations":[{"path":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:python-jinja2:python-jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python_jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python-jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python_jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python-jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python_jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python-jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python_jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python:jinja2:3.1.3:*:*:*:*:*:*:*"],"purl":"pkg:pypi/jinja2@3.1.3","upstreams":[]}},{"vulnerability":{"id":"GHSA-gmj6-6f8f-6699","dataSource":"https://github.com/advisories/GHSA-gmj6-6f8f-6699","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Jinja has a sandbox breakout through malicious filenames","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:N/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:H","metrics":{"baseScore":8.8,"exploitabilityScore":2.9,"impactScore":5.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-56201","epss":0.00064,"percentile":0.2041,"date":"2025-06-03"}],"fix":{"versions":["3.1.5"],"state":"fixed"},"advisories":[],"risk":0.044160000000000005},"relatedVulnerabilities":[{"id":"CVE-2024-56201","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2024-56201","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pallets/jinja/commit/767b23617628419ae3709ccfb02f9602ae9fe51f","https://github.com/pallets/jinja/issues/1792","https://github.com/pallets/jinja/releases/tag/3.1.5","https://github.com/pallets/jinja/security/advisories/GHSA-gmj6-6f8f-6699"],"description":"Jinja is an extensible templating engine. In versions on the 3.x branch prior to 3.1.5, a bug in the Jinja compiler allows an attacker that controls both the content and filename of a template to execute arbitrary Python code, regardless of if Jinja's sandbox is used. To exploit the vulnerability, an attacker needs to control both the filename and the contents of a template. Whether that is the case depends on the type of application using Jinja. This vulnerability impacts users of applications which execute untrusted templates where the template author can also choose the template filename. This vulnerability is fixed in 3.1.5.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:L/AC:L/AT:P/PR:L/UI:P/VC:H/VI:H/VA:H/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":5.4},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2024-56201","epss":0.00064,"percentile":0.2041,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"jinja2","version":"3.1.3"}},"found":{"versionConstraint":">=3.0.0,<=3.1.4 (python)","vulnerabilityID":"GHSA-gmj6-6f8f-6699"},"fix":{"suggestedVersion":"3.1.5"}}],"artifact":{"id":"8ef17ffb7cb9e30e","name":"jinja2","version":"3.1.3","type":"python","locations":[{"path":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:python-jinja2:python-jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python_jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python-jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python_jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python-jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python_jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python-jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python_jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python:jinja2:3.1.3:*:*:*:*:*:*:*"],"purl":"pkg:pypi/jinja2@3.1.3","upstreams":[]}},{"vulnerability":{"id":"GHSA-cpwx-vrp4-4pq7","dataSource":"https://github.com/advisories/GHSA-cpwx-vrp4-4pq7","namespace":"github:language:python","severity":"Medium","urls":[],"description":"Jinja2 vulnerable to sandbox breakout through attr filter selecting format method","cvss":[],"epss":[{"cve":"CVE-2025-27516","epss":0.00034,"percentile":0.08481,"date":"2025-06-03"}],"fix":{"versions":["3.1.6"],"state":"fixed"},"advisories":[],"risk":0.017},"relatedVulnerabilities":[{"id":"CVE-2025-27516","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2025-27516","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pallets/jinja/commit/90457bbf33b8662926ae65cdde4c4c32e756e403","https://github.com/pallets/jinja/security/advisories/GHSA-cpwx-vrp4-4pq7","https://lists.debian.org/debian-lts-announce/2025/04/msg00045.html"],"description":"Jinja is an extensible templating engine. Prior to 3.1.6, an oversight in how the Jinja sandboxed environment interacts with the |attr filter allows an attacker that controls the content of a template to execute arbitrary Python code. To exploit the vulnerability, an attacker needs to control the content of a template. Whether that is the case depends on the type of application using Jinja. This vulnerability impacts users of applications which execute untrusted templates. Jinja's sandbox does catch calls to str.format and ensures they don't escape the sandbox. However, it's possible to use the |attr filter to get a reference to a string's plain format method, bypassing the sandbox. After the fix, the |attr filter no longer bypasses the environment's attribute lookup. This vulnerability is fixed in 3.1.6.","cvss":[{"source":"security-advisories@github.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:L/AC:L/AT:P/PR:L/UI:P/VC:H/VI:H/VA:H/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":5.4},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-27516","epss":0.00034,"percentile":0.08481,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"jinja2","version":"3.1.3"}},"found":{"versionConstraint":"<=3.1.5 (python)","vulnerabilityID":"GHSA-cpwx-vrp4-4pq7"},"fix":{"suggestedVersion":"3.1.6"}}],"artifact":{"id":"8ef17ffb7cb9e30e","name":"jinja2","version":"3.1.3","type":"python","locations":[{"path":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/Jinja2-3.1.3.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3-Clause"],"cpes":["cpe:2.3:a:python-jinja2:python-jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:python_jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python-jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:python_jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python-jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:python_jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python-jinja2:jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python-jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python:python_jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python_jinja2:jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:jinja2:jinja2:3.1.3:*:*:*:*:*:*:*","cpe:2.3:a:python:jinja2:3.1.3:*:*:*:*:*:*:*"],"purl":"pkg:pypi/jinja2@3.1.3","upstreams":[]}},{"vulnerability":{"id":"GHSA-887c-mr87-cxwp","dataSource":"https://github.com/advisories/GHSA-887c-mr87-cxwp","namespace":"github:language:python","severity":"Medium","urls":[],"description":"PyTorch Improper Resource Shutdown or Release vulnerability","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:L","metrics":{"baseScore":3.3,"exploitabilityScore":1.9,"impactScore":1.5},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-3730","epss":0.00022,"percentile":0.04435,"date":"2025-06-03"}],"fix":{"versions":[],"state":"not-fixed"},"advisories":[],"risk":0.00913},"relatedVulnerabilities":[{"id":"CVE-2025-3730","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2025-3730","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pytorch/pytorch/issues/150835","https://github.com/pytorch/pytorch/issues/150835#issue-2979082232","https://github.com/pytorch/pytorch/pull/150981","https://github.com/timocafe/tewart-pytorch/commit/46fc5d8e360127361211cb237d5f9eef0223e567","https://vuldb.com/?ctiid.305076","https://vuldb.com/?id.305076","https://vuldb.com/?submit.553645","https://github.com/pytorch/pytorch/issues/150835"],"description":"A vulnerability, which was classified as problematic, was found in PyTorch 2.6.0. Affected is the function torch.nn.functional.ctc_loss of the file aten/src/ATen/native/LossCTC.cpp. The manipulation leads to denial of service. An attack has to be approached locally. The exploit has been disclosed to the public and may be used. The real existence of this vulnerability is still doubted at the moment. The name of the patch is 46fc5d8e360127361211cb237d5f9eef0223e567. It is recommended to apply a patch to fix this issue. The security policy of the project warns to use unknown models which might establish malicious effects.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":5.5,"exploitabilityScore":1.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"cna@vuldb.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:L/AC:L/AT:N/PR:L/UI:N/VC:N/VI:N/VA:L/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":4.8},"vendorMetadata":{}},{"source":"cna@vuldb.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:L","metrics":{"baseScore":3.3,"exploitabilityScore":1.9,"impactScore":1.5},"vendorMetadata":{}},{"source":"cna@vuldb.com","type":"Secondary","version":"2.0","vector":"AV:L/AC:L/Au:S/C:N/I:N/A:P","metrics":{"baseScore":1.7,"exploitabilityScore":3.2,"impactScore":2.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-3730","epss":0.00022,"percentile":0.04435,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"torch","version":"2.4.1+cpu"}},"found":{"versionConstraint":"<=2.7.0 (python)","vulnerabilityID":"GHSA-887c-mr87-cxwp"}}],"artifact":{"id":"c15427c9f84b45c1","name":"torch","version":"2.4.1+cpu","type":"python","locations":[{"path":"/lib/python3.10/site-packages/torch-2.4.1+cpu.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/torch-2.4.1+cpu.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/torch-2.4.1+cpu.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/torch-2.4.1+cpu.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/torch-2.4.1+cpu.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/torch-2.4.1+cpu.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3"],"cpes":["cpe:2.3:a:pytorch_team_project:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_team_project:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_teamproject:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_teamproject:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packages_project:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packages_project:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packagesproject:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packagesproject:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_team_project:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python-torch:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python-torch:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python_torch:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python_torch:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_team:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_team:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_teamproject:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packages_project:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packages:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packages:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packagesproject:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python-torch:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python_torch:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_team:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:torch:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:torch:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packages:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:torch:torch:2.4.1+cpu:*:*:*:*:*:*:*"],"purl":"pkg:pypi/torch@2.4.1%2Bcpu","upstreams":[]}},{"vulnerability":{"id":"GHSA-3749-ghw9-m3mg","dataSource":"https://github.com/advisories/GHSA-3749-ghw9-m3mg","namespace":"github:language:python","severity":"Low","urls":[],"description":"PyTorch susceptible to local Denial of Service","cvss":[{"type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:L","metrics":{"baseScore":3.3,"exploitabilityScore":1.9,"impactScore":1.5},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-2953","epss":0.00023,"percentile":0.04624,"date":"2025-06-03"}],"fix":{"versions":["2.7.1-rc1"],"state":"fixed"},"advisories":[],"risk":0.007244999999999999},"relatedVulnerabilities":[{"id":"CVE-2025-2953","dataSource":"https://nvd.nist.gov/vuln/detail/CVE-2025-2953","namespace":"nvd:cpe","severity":"Medium","urls":["https://github.com/pytorch/pytorch/blob/main/SECURITY.md#untrusted-models","https://github.com/pytorch/pytorch/issues/149274","https://github.com/pytorch/pytorch/issues/149274#issue-2923122269","https://vuldb.com/?ctiid.302006","https://vuldb.com/?id.302006","https://vuldb.com/?submit.521279","https://github.com/pytorch/pytorch/issues/149274"],"description":"A vulnerability, which was classified as problematic, has been found in PyTorch 2.6.0+cu124. Affected by this issue is the function torch.mkldnn_max_pool2d. The manipulation leads to denial of service. An attack has to be approached locally. The exploit has been disclosed to the public and may be used. The real existence of this vulnerability is still doubted at the moment. The security policy of the project warns to use unknown models which might establish malicious effects.","cvss":[{"source":"nvd@nist.gov","type":"Primary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:H","metrics":{"baseScore":5.5,"exploitabilityScore":1.9,"impactScore":3.6},"vendorMetadata":{}},{"source":"cna@vuldb.com","type":"Secondary","version":"4.0","vector":"CVSS:4.0/AV:L/AC:L/AT:N/PR:L/UI:N/VC:N/VI:N/VA:L/SC:N/SI:N/SA:N/E:X/CR:X/IR:X/AR:X/MAV:X/MAC:X/MAT:X/MPR:X/MUI:X/MVC:X/MVI:X/MVA:X/MSC:X/MSI:X/MSA:X/S:X/AU:X/R:X/V:X/RE:X/U:X","metrics":{"baseScore":4.8},"vendorMetadata":{}},{"source":"cna@vuldb.com","type":"Secondary","version":"3.1","vector":"CVSS:3.1/AV:L/AC:L/PR:L/UI:N/S:U/C:N/I:N/A:L","metrics":{"baseScore":3.3,"exploitabilityScore":1.9,"impactScore":1.5},"vendorMetadata":{}},{"source":"cna@vuldb.com","type":"Secondary","version":"2.0","vector":"AV:L/AC:L/Au:S/C:N/I:N/A:P","metrics":{"baseScore":1.7,"exploitabilityScore":3.2,"impactScore":2.9},"vendorMetadata":{}}],"epss":[{"cve":"CVE-2025-2953","epss":0.00023,"percentile":0.04624,"date":"2025-06-03"}]}],"matchDetails":[{"type":"exact-direct-match","matcher":"python-matcher","searchedBy":{"language":"python","namespace":"github:language:python","package":{"name":"torch","version":"2.4.1+cpu"}},"found":{"versionConstraint":"<2.7.1-rc1 (python)","vulnerabilityID":"GHSA-3749-ghw9-m3mg"},"fix":{"suggestedVersion":"2.7.1-rc1"}}],"artifact":{"id":"c15427c9f84b45c1","name":"torch","version":"2.4.1+cpu","type":"python","locations":[{"path":"/lib/python3.10/site-packages/torch-2.4.1+cpu.dist-info/METADATA","accessPath":"/lib/python3.10/site-packages/torch-2.4.1+cpu.dist-info/METADATA","annotations":{"evidence":"primary"}},{"path":"/lib/python3.10/site-packages/torch-2.4.1+cpu.dist-info/RECORD","accessPath":"/lib/python3.10/site-packages/torch-2.4.1+cpu.dist-info/RECORD","annotations":{"evidence":"supporting"}},{"path":"/lib/python3.10/site-packages/torch-2.4.1+cpu.dist-info/top_level.txt","accessPath":"/lib/python3.10/site-packages/torch-2.4.1+cpu.dist-info/top_level.txt","annotations":{"evidence":"supporting"}}],"language":"python","licenses":["BSD-3"],"cpes":["cpe:2.3:a:pytorch_team_project:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_team_project:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_teamproject:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_teamproject:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packages_project:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packages_project:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packagesproject:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packagesproject:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_team_project:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python-torch:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python-torch:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python_torch:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python_torch:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_team:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_team:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_teamproject:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packages_project:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packages:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packages:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packagesproject:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python-torch:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python_torch:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:pytorch_team:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:torch:python-torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:torch:python_torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:packages:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:python:torch:2.4.1+cpu:*:*:*:*:*:*:*","cpe:2.3:a:torch:torch:2.4.1+cpu:*:*:*:*:*:*:*"],"purl":"pkg:pypi/torch@2.4.1%2Bcpu","upstreams":[]}}],"source":{"type":"sbom-file","target":"whisper-gael-syft.json"},"distro":{"name":"","version":"","idLike":null},"descriptor":{"name":"grype","version":"0.92.2","configuration":{"output":["json=/Users/alan/Personal/popey/syft-snap-source-test/grype/v1.26.0/2025-06-04/whisper-gael-grype.json","template=/Users/alan/Personal/popey/syft-snap-source-test/grype/v1.26.0/2025-06-04/whisper-gael-grype.html"],"file":"","pretty":false,"distro":"","add-cpes-if-none":false,"output-template-file":"../html.tmpl","check-for-app-update":true,"only-fixed":false,"only-notfixed":false,"ignore-wontfix":"","platform":"","search":{"scope":"squashed","unindexed-archives":false,"indexed-archives":true},"ignore":[{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"kernel-headers","version":"","language":"","type":"rpm","location":"","upstream-name":"kernel"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"},{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"linux(-.*)?-headers-.*","version":"","language":"","type":"deb","location":"","upstream-name":"linux.*"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"},{"vulnerability":"","reason":"","namespace":"","fix-state":"","package":{"name":"linux-libc-dev","version":"","language":"","type":"deb","location":"","upstream-name":"linux"},"vex-status":"","vex-justification":"","match-type":"exact-indirect-match"}],"exclude":[],"externalSources":{"enable":false,"maven":{"searchUpstreamBySha1":true,"baseUrl":"https://search.maven.org/solrsearch/select","rateLimit":300000000}},"match":{"java":{"using-cpes":false},"jvm":{"using-cpes":true},"dotnet":{"using-cpes":false},"golang":{"using-cpes":false,"always-use-cpe-for-stdlib":true,"allow-main-module-pseudo-version-comparison":false},"javascript":{"using-cpes":false},"python":{"using-cpes":false},"ruby":{"using-cpes":false},"rust":{"using-cpes":false},"stock":{"using-cpes":true}},"fail-on-severity":"","registry":{"insecure-skip-tls-verify":false,"insecure-use-http":false,"auth":null,"ca-cert":""},"show-suppressed":false,"by-cve":false,"SortBy":{"sort-by":"risk"},"name":"","default-image-pull-source":"","vex-documents":[],"vex-add":[],"match-upstream-kernel-headers":false,"db":{"cache-dir":"/Users/alan/Library/Caches/grype/db","update-url":"https://grype.anchore.io/databases","ca-cert":"","auto-update":true,"validate-by-hash-on-start":true,"validate-age":true,"max-allowed-built-age":432000000000000,"require-update-check":false,"update-available-timeout":30000000000,"update-download-timeout":300000000000,"max-update-check-frequency":7200000000000},"exp":{},"dev":{"db":{"debug":false}}},"db":{"status":{"schemaVersion":"v6.0.2","from":"https://grype.anchore.io/databases/v6/vulnerability-db_v6.0.2_2025-06-04T01:32:23Z_1749035557.tar.zst?checksum=sha256%3A77238f4ffbc7cf72bf4dedabbb21a3d0dd3ade90fb5d9df7c4da2870f8d1a9b7","built":"2025-06-04T11:12:37Z","path":"/Users/alan/Library/Caches/grype/db/6/vulnerability.db","valid":true},"providers":{"alpine":{"captured":"2025-06-04T01:32:23Z","input":"xxh64:a6a065ece392f6d1"},"amazon":{"captured":"2025-06-04T01:32:32Z","input":"xxh64:b65d458d6858d72a"},"chainguard":{"captured":"2025-06-04T01:32:33Z","input":"xxh64:03c9c4d182d77791"},"debian":{"captured":"2025-06-04T01:32:49Z","input":"xxh64:169ceb8fbcb8d6f1"},"epss":{"captured":"2025-06-04T01:32:42Z","input":"xxh64:bf87c4c234d02346"},"github":{"captured":"2025-06-04T01:32:30Z","input":"xxh64:3dab0be50635e057"},"kev":{"captured":"2025-06-04T01:32:45Z","input":"xxh64:a934e75f468f4e09"},"mariner":{"captured":"2025-06-04T01:32:43Z","input":"xxh64:ac698752085c186f"},"nvd":{"captured":"2025-06-04T01:33:28Z","input":"xxh64:389887843a59d29c"},"oracle":{"captured":"2025-06-04T01:32:35Z","input":"xxh64:0b8aabcfb6505b62"},"rhel":{"captured":"2025-06-04T04:49:10Z","input":"xxh64:947d0caeb19e9ccc"},"sles":{"captured":"2025-06-04T01:32:34Z","input":"xxh64:1c50b1219296628a"},"ubuntu":{"captured":"2025-06-04T01:33:14Z","input":"xxh64:71e22d5c5bbe7d14"},"wolfi":{"captured":"2025-06-04T01:32:42Z","input":"xxh64:e68fca5a870c6769"}}},"timestamp":"2025-06-04T19:51:25.227624+01:00"}}
